docs/testing/webdriverio/01-overview.md

```diff
@@ -0,0 +1,35 @@
+---
+title: WebdriverIO Overview
+sidebar_label: Overview
+---
+
+# Overview
+
+WebdriverIO is a progressive automation framework built to automate modern web and mobile applications. It simplifies the interaction with your app and provides a set of plugins that help you create a scalable, robust and stable test suite.
+
+Testing with WebdriverIO has the following advantages:
+
+- __Cross Browser Support__: WebdriverIO is designed to support all platforms, either on desktop or mobile. You can run tests on actual browser your users are using, including covering different versions of them.
+- __Real User Interaction__: Interacting with elements in WebdriverIO through the WebDriver protocol is the closes you can get to reality compared to emulated interaction in virtual DOM environments.
+- __Web Platform Support__: Running tests in actual browser allows you to tap into the latest Web Platform features for testing your components, often not available when using virtual DOM environments.
+
+## Set Up
+
+To get started with WebdriverIO, all you need to do is to run their project starter:
+
+```sh
+npm init wdio@latest .
+```
+
+This will initiate WebdriverIOs configuration wizard that walks you through the setup. Make sure you select the following options when walking through it:
+
+- __What type of testing would you like to do?__: select either `Component or Unit Testing - in the browser` if you are interested adding unit tests for your components or `E2E Testing - of Web or Mobile Applications` if you like to test your whole application (you can always add either of them later on)
+- __Which framework do you use for building components?__: if you select _Component or Unit Testing_ make sure to select `StencilJS` as preset so WebdriverIO knows how to compile your components properly
+
+The following questions can be answered as desired. Once setup the wizard has created a `wdio.conf.ts` file and a `wdio` script to run your tests. You should be able to run your first test on the auto-generated test file via:
+
+```sh
+npm run wdio
+```
+
+More information on setting up WebdriverIO can be found in their [documentation](https://webdriver.io/docs/component-testing/stencil).
```

Comment:
I was able to stand up a new project easily with `npm init wdio@latest` and so on and faced no issues, however when I tried to _add_ webdriverio stuff to an existing project I faced a bit of an issue.

I first created a new test stencil project like so:

```sh
mkdir /tmp/test && cd /tmp/test
npm init stencil component test-adding-wdio
```

then after doing an `npm install` and whatnot I ran:

```sh
npm init wdio@latest .
```

this ran through the wizard with no problems, and it looks like it cleanly installed things over my existing setup, but then when I try `npm run wdio` I get:

<img width="999" alt="Screenshot 2024-02-07 at 2 26 36 PM" src="https://github.com/ionic-team/stencil-site/assets/6207644/85b14b92-ed34-405a-aed3-657d4fc2c0a0">

I then went to install `ts-node` and then get a different error:

<img width="1000" alt="Screenshot 2024-02-07 at 2 27 26 PM" src="https://github.com/ionic-team/stencil-site/assets/6207644/dd3ce1cd-f145-4054-92ca-2bfa40a5d05f">

I have no idea what I'm doing with wdio so probably missing something! But j/w two things:

- is adding wdio tests to an existing project using the `init` starter something that's supported?
- if so, is that something we should document here?

I imagine there's going to be some interest in using this approach for the e2e tests in existing projects, so we might want to make sure that's documented.

Reply:
Two errors happened here which are bad DX and a bug in WebdriverIO:

- first, currently WebdriverIO assumes `ts-node` to be installed for all TypeScript projects, this assumption is wrong though. I [raised an issue](https://github.com/webdriverio/webdriverio/issues/12197) in the WebdriverIO project and can tackle this quickly.
- second, the config is written in ESM while a new Stencil project is set up as CJS project - it is currently very difficult to find out what environment the project is set up with but this is DX I can also fix on the WebdriverIO side - in the meantime, let me add a comment to rename the config to `.mts`

Reply:
I ran into this as well - would renaming the config extension to `.mts` work? I tried that locally and ran into issues there:
```
ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///private/tmp/wdio-basic-test/wdio.conf.ts: Cannot find module '/private/tmp/wdio-basic-test/wdio.conf.ts' imported from /private/tmp/wdio-basic-test/node_modules/@wdio/config/build/node/RequireLibrary.js
Error: Cannot find module '/private/tmp/wdio-basic-test/wdio.conf.ts' imported from /private/tmp/wdio-basic-test/node_modules/@wdio/config/build/node/RequireLibrary.js
    at finalizeResolution (/private/tmp/wdio-basic-test/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:366:11)
    at moduleResolve (/private/tmp/wdio-basic-test/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:801:10)
    at Object.defaultResolve (/private/tmp/wdio-basic-test/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:912:11)
    at /private/tmp/wdio-basic-test/node_modules/ts-node/src/esm.ts:218:35
    at entrypointFallback (/private/tmp/wdio-basic-test/node_modules/ts-node/src/esm.ts:168:34)
    at /private/tmp/wdio-basic-test/node_modules/ts-node/src/esm.ts:217:14
    at addShortCircuitFlag (/private/tmp/wdio-basic-test/node_modules/ts-node/src/esm.ts:409:21)
    at resolve (/private/tmp/wdio-basic-test/node_modules/ts-node/src/esm.ts:197:12)
    at nextResolve (node:internal/modules/esm/hooks:865:28)
    at Hooks.resolve (node:internal/modules/esm/hooks:303:30)
```

Wouldn't the config file need to be a CJS module? Or am I misunderstanding the issue here?

Reply:
@rwaskiewicz I think you missed to update the `wdio` script in the `package.json` which still points to a `wdio.conf.ts` file. I tested it and changing to `.mts` works. The problem is that the Stencil starter project is a CJS project unless you do either of the two changes:

- rename `wdio.conf.ts` to `wdio.conf.mts` (my preferred choice)
- add `"type": "module"` to `package.json` (related to work proposed in STENCIL-1051)

The other option is to have WebdriverIO compose the whole thing as CJS (use `require` instead `import`) but that sounds like a step backwards to me.

I will add an admonition telling users to make this change, and check the ESM state of their project as I think this is the easiest way to unblock this.

Reply:
I added the following two admonitions:

<img width="754" alt="Screenshot 2024-02-08 at 6 27 20 PM" src="https://github.com/ionic-team/stencil-site/assets/731337/30dcd16b-c69c-4057-a636-d8df5a246dc0">

The example component generated by WebdriverIO uses fragments which causes TypeScript to fail due to missing `jsxFragmentFactory` settings. I left this out as it seems straightforward how to fix it and chances are that users either don't generate them or remove them immediately. Let me know if I should add an admonition for this too.